### PR TITLE
Removing use of `BaseObject` constructor

### PR DESCRIPTION
**Pull request by larryli**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | no

### CODE REVIEW

File: src/DataReader.php [hunk 1]
```diff
@@ -68,7 +68,6 @@ public function __construct(Command $command, $config = [])
     {
         $this->_statement = $command->pdoStatement;
         $this->_statement->setFetchMode(\PDO::FETCH_ASSOC);
-        parent::__construct($config);
```
Comment (cebe) [inline src/DataReader.php]: you still have `$config` in the constructor signature.

Btw, can someone point me to the discussion, on why this has been removed from Yii core?

Comment (larryli) [reply src/DataReader.php]: I don't know.
Just fix "Error: Cannot call constructor".

**Comment by kids-return** [comment]
Hi @cebe 
The reason is here
https://github.com/yiisoft/yii-core/commit/a16ba215994336fcc50e7340824af3e102df7fe2#diff-828ad3929c2c6e3ba47c513ec3625631

**Comment by machour** [comment]
Closing this as it was handled in https://github.com/yiisoft/db/commit/41fafcfd606deb453988112c5ec2786793487b2d

**Pull request closed without merging**